tools/regs2groov.py

```diff
@@ -0,0 +1,134 @@
+import xml.etree.ElementTree as et
+import re
+from collections import namedtuple
+import argparse
+
+
+Group = namedtuple('Group', ['name', 'registers'])
+Register = namedtuple('Register', ['name', 'address', 'fields'])
+Field = namedtuple('Field', ['name', 'access', 'msb', 'lsb'])
+
+def parse_svd(filename):
+    def mk_field(x):
+        (msb, lsb) = re.search(r"\[(\d+):(\d+)\]", x.find('bitRange').text).groups()
+
+        return Field(
+            name = x.find('name').text,
+            access = x.find('access').text,
+            msb = int(msb),
+            lsb = int(lsb)
+        )
+
+    def mk_register(x, base):
+        return Register(
+            name = x.find('name').text,
+            address = base + int(x.find('addressOffset').text[2:], 16),
+            fields = [mk_field(f) for f in x.findall('.//field')]
+        )
+
+    def mk_group(x, root):
+        base_addr = int(x.find('baseAddress').text[2:], 16)
+        name = x.find('name').text
+
+        if ("derivedFrom" in x.attrib):
+            x = root.find(f""".//peripheral[name='{x.attrib["derivedFrom"]}']""")
+            
+        return Group(
+            name = name,
+            registers = [mk_register(r, base_addr) for r in x.findall('.//register')]
+        )
+    
+    root = et.parse(filename).getroot()
+    return [mk_group(p, root) for p in root.findall('.//peripheral')]
+
+
+def generate_groups(groups, dest, namespace, name_func):
+    def indent(lines, len=4):
```

Comment:
**thought**: We are accumulating manual indents in a few places - I'd like to see if we can use textwrap or some other systemic solution that we don't need to maintain.

Reply:
Maybe we just don't worry about formatting the code and let clang-format do it? Just one big line...